include/hipSYCL/sycl/libkernel/marray.hpp

```diff
@@ -104,7 +92,9 @@ template <typename DataT, std::size_t NumElements> class marray {
   constexpr marray(marray<DataT, NumElements>&& rhs) = default;
 
   // Available only when: NumElements == 1
-  operator scalar_conversion_type() const {
+  template <std::size_t N = NumElements,
+            std::enable_if_t<N == 1, bool> = true>
+  constexpr operator DataT() const {
```

Comment:
This change might imply a semantic change. In the old implementation, the following should work:

```
void f(int x) {....}

marray<int, 1> x = ...;
f(x);
```
which I'm not sure will continue to work after the change because implicit conversion rules are different for templates.

I remember we had lots of issues getting the implicit conversion from `id<1>` to `size_t` to work, and this was the solution.

Here it might be worth checking what DPC++ does to ensure compatibility.

Reply:
IIRC I did check what DPC++ does and I just checked again, and they are doing the same as we do now. Your example seems to still work, but just to be sure, I'll add a test case.

Reply:
Ok, if DPC++ also handles it this way, that's fine.